specification/quota/resource-manager/Microsoft.Quota/preview/2023-06-01-preview/groupquota.json

```diff
@@ -0,0 +1,1797 @@
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{mgId}/providers/Microsoft.Quota/groupQuotas/{groupQuotaName}/subscriptions/{subscriptionId}": {
+      "put": {
+        "tags": [
+          "GroupQuotas-Subscriptions"
+        ],
+        "summary": "Adds subscription to the GroupQuotas.",
+        "description": "Adds a subscription to GroupQuotas.",
+        "operationId": "GroupQuotaSubscriptions_CreateOrUpdate",
+        "x-ms-examples": {
+          "GroupQuotaSubscriptions_Put_Subscriptions": {
+            "$ref": "./examples/GroupQuotas-Subscriptions/PutGroupQuotasSubscription.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/MgIdInParameters"
+          },
+          {
+            "$ref": "#/parameters/GroupQuotaNameInParameters"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK, Create or Update",
+            "schema": {
+              "$ref": "#/definitions/GroupQuotasEntity"
+            }
+          },
+          "201": {
+            "description": "Request Accepted, Create or Update",
+            "headers": {
+              "Retry-After": {
+                "description": "The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.",
+                "type": "integer",
+                "format": "int64"
+              },
+              "Location": {
+                "description": "URL for determining when an operation has completed. Send a GET request to the URL in Location header.\nThe URI should return a 202 until the operation reaches a terminal state and 200 once it reaches a terminal state.\n\nFor more info: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#202-accepted-and-location-headers.  New guidelines - https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#131-resource-based-long-running-operations-relo",
+                "type": "string"
+              },
+              "Azure-AsyncOperation": {
+                "description": "URL for checking the ongoing status of the operation.\nTo get the status of the asynchronous operation, send a GET request to the URL in Azure-AsyncOperation header value.\n\nFor more info: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/OperationsStatus"
```

Comment:
>OperationsStatus [](http://example.com/codeflow?start=37&length=16)

The 200 response should also return GroupQuotasEntity, not OperationStatus. The Azure-AsyncOperation header returns the URL, making a GET to that URL, returns OperationStatus as response. See: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md).

Reply:
<img width="471" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/67396382/99c91f43-43fc-41ff-924e-ff7b9ba5bd68">
is this not correct? from what I can see, the `OperationStatus` here is returned for the 201

Reply:
We can add a 200 return, as ARM requires 200 responses for PUT, Patch.

Reply:
the screenshot i've attached above seems to be the 200 response for this PUT

Reply:
With 201 it should return the same schema with the appropriate ProvisioningState, and user will use the header to check the status. Here it should still return GroupQuotasEntity, see [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

Reply:
updated put 201 returns. let me know if further action is required